### PR TITLE
LPS-140723 Fix the workflow version in the xml to update

### PR DESCRIPTION
**Pull request by MariMalta**

### CODE REVIEW

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/legacy/xml_definition_serializer.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ AUI.add(
 		var serializeDefinition = function (xmlNamespace, metadata, json) {
 			var description = metadata.description;
 			var name = metadata.name;
-			var version = metadata.version;
+			var version = parseInt(metadata.version, 10);
```
Comment (rafaprax) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/legacy/xml_definition_serializer.js:39]: @MariMalta @MarinhoFeliphe Just a question, do we really need to convert this here? Thx

**Review by rafaprax — CHANGES_REQUESTED** [review]

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/legacy/xml_definition_serializer.js (hunk 1 above)
Comment (MariMalta) [reply modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/legacy/xml_definition_serializer.js:39]: I don't think we really need to, but in the xml right now the version is 1 and not 1.0, that's why the parseInt is there.

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/legacy/main.js [hunk 2]
```diff
@@ -444,10 +444,20 @@ AUI.add(
 				getContent() {
 					var instance = this;
 
+					const portletNamespace =
+						instance._state.data.portletNamespace;
+
```
[the remark this answers is not in the document]
Comment (MariMalta) [reply modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/legacy/main.js:449]: Okay, I'll check it

**Comment by rafaprax** [comment]
@MarinhoFeliphe @MariMalta 

> LocalFile.Kaleodesigner#ViewSourceXMLNotErasedWhenInvalid - Poshi Report - Poshi Summary - Console Output
LocalFile.Kaleodesigner#PublishDefaultWorkflowDefinition - Poshi Report - Poshi Summary - Console Output

These two tests have been failed in the last 2 runs. So, probably these failures a related to these changes somehow. Can you check with the QA guys?

Thx

**Comment by MariMalta** [comment]
ci:test:relevant

**Comment by MariMalta** [comment]
ci:test:relevant

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/legacy/xml_definition_serializer.js (hunk 1 above)
Comment (MariMalta) [reply modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/legacy/xml_definition_serializer.js:39]: I've tried with the 1.0 and doesn't work. Without the parseInt we receive an alert to enter a valid content.

**Comment by MariMalta** [comment]
ci:test:relevant

**Comment by MariMalta** [comment]
ci:test:relevant

**Review by MarinhoFeliphe — APPROVED** [review]

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/legacy/definition_diagram_controller.js [hunk 3]
```diff
@@ -123,16 +123,20 @@ AUI.add(
 				return fields;
 			},
 
-			serializeDefinition(json) {
+			serializeDefinition(json, draftVersion) {
```
Comment (rafaprax) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/legacy/definition_diagram_controller.js:126]: Hey @MariMalta, Can you please sort the parameters alphabetically? Thx

**Review by rafaprax — CHANGES_REQUESTED** [review]

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/legacy/definition_diagram_controller.js (hunk 3 above)
Comment (MariMalta) [reply modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/legacy/definition_diagram_controller.js:126]: Yes, I'm going to do it

**Comment by rafaprax** [comment]
ci:forward

**Comment by rafaprax** [comment]
ci:forward

**Comment by rafaprax** [comment]
ci:forward

**Review by rafaprax — APPROVED** [review]

**Comment by rafaprax** [comment]
ci:forward

**Comment by rafaprax** [comment]
Hey @anabuchmann, Can you please check if there are related failures here and forward this pr manually to BChan if not?

Thx

**Comment by anabuchmann** [comment]
Just started reviewing :)

**Comment by MariMalta** [comment]
ci:test:sf

**Comment by MariMalta** [comment]
ci:test:relevant

**Comment by MariMalta** [comment]
Just started reviewing :)

**Comment by lucasperj** [comment]
Just started reviewing :)

**Comment by lucasperj** [comment]
Manually forwarded here: [pull/109848](https://github.com/brianchandotcom/liferay-portal/pull/109848). Thank you!

**Pull request closed without merging**